lua/curl/shell_utils.lua

```diff
@@ -0,0 +1,55 @@
+local M = {}
+
+local cache = {}
+
+M.has_command = function(cmd)
+	return vim.fn.executable(cmd) == 1
+end
+
+M.has_powershell = function()
+	if cache.has_powershell ~= nil then
+		return cache._has_powershell
+	end
+	cache.has_powershell = M.has_command("powershell")
+	return cache.has_powershell
+end
+
+M.has_sh = function()
+	if cache.has_sh ~= nil then
+		return cache.has_sh
+	end
+	cache.has_sh = M.has_command("sh")
+	return cache.has_sh
+end
+
+M.has_bash = function()
+	if cache.has_bash ~= nil then
+		return cache.has_bash
+	end
+	cache.has_bash = M.has_command("bash")
+	return cache.has_bash
+end
+
+M.has_zsh = function()
+	if cache.has_zsh ~= nil then
+		return cache.has_zsh
+	end
+	cache.has_zsh = M.has_command("zsh")
+	return cache.has_zsh
+end
+
+M.get_default_shell = function()
+	if M.has_bash() then
+		return { "bash", "-c" }
+	elseif M.has_sh() then
+		return { "sh", "-c" }
+	elseif M.has_zsh() then
+		return { "zsh", "-c" }
+	elseif M.has_powershell() then
+		return { "powershell", "-Command" }
+	else
+		error("Shell not found: bash, sh, zsh, powershell.")

```

Comment:
Can we just return an empty table here, so that we do a best effort attempt to run the command if a shell is not found?

Reply:
I tried the `{"curl xxxx"}` format and it doesn't seem to work.

Reply:
I thought it would be nice to highlight some of the shell notes in the README before the refactoring is complete.

Reply:
Could we maybe say that if there's no found shell, it reverts to the previous behaviour? 
I really don't want to have the plugin break for people because their shell is not defined in our code, when we don't actually need do anything for it to work.

Also, I dont think this plugin works with powershell since it relies on JQ, which I dont think can be installed in Powershell. This might give the impression that powershell is supported.